### PR TITLE
Trwalke/selenium tests

### PR DESCRIPTION
**Pull request by trwalke**
Migrating the selenium test infrastructure to ADAL.
This PR contains 6 new Selenium desktop tests. The rest will be added later.

### CODE REVIEW

File: devApps/XFormsApp/SecondPage.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public class SecondPage : ContentPage
 
         public string User = "<User>";
         private string Tenant = "<Tenant>";
-        public const string AndroidBrokerRedirectURI = "msauth://com.microsoft.xformsdroid.adal/mJaAVvdXtcXy369xPWv2C7mV674=";
+        public const string AndroidBrokerRedirectURI = "adaliosapp://com.yourcompany.xformsapp";
```
Comment (henrik-me) [inline devApps/XFormsApp/SecondPage.cs]: >adaliosapp [](start = 56, length = 10)

nit: seems off to call the protocol ios app for android. 
Is it actually used for testing?

File: tests/Test.ADAL.Integration/Infrastructure/SeleniumExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Support.UI;
+using System.Diagnostics;
+using Test.Microsoft.Identity.LabInfrastructure;
+using Test.Microsoft.Identity.Core.UIAutomation;
+
+namespace Test.ADAL.Integration.Infrastructure
+{
+    public static class SeleniumExtensions
```
Comment (henrik-me) [inline tests/Test.ADAL.Integration/Infrastructure/SeleniumExtensions.cs:40]: Consider making test infrastructure and test helpers which are used both for ADAL and MSAL ensuring both are updated if we make changes.

File: tests/Test.ADAL.Integration/Infrastructure/SingleMessageTcpListener.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            //string message = null;
+            //switch (result.Status)
+            //{
+            //    case AuthorizationStatus.Success:
+            //        message = CloseWindowSuccessHtml;
+            //        break;
+            //    default:
+            //        message = CloseWindowFailureHtml;
+            //        break;
+            //}
```
Comment (henrik-me) [inline tests/Test.ADAL.Integration/Infrastructure/SingleMessageTcpListener.cs]: ?

File: tests/Test.Microsoft.Identity.Core.UIAutomation/CoreMobileTestHelper.cs [hunk 4]
```diff
@@ -102,13 +101,15 @@ private void RetryVerificationHelper(Action verification)
                 catch (ResultVerificationFailureException ex)
                 {
                     if (attempts == CoreUiTestConstants.MaximumResultCheckRetryAttempts)
-                        Assert.Fail("Could not Verify test result");
+                        throw ex;
+                        //("Could not Verify test result");
```
Comment (henrik-me) [inline tests/Test.Microsoft.Identity.Core.UIAutomation/CoreMobileTestHelper.cs]: Why move away from asserts and using exceptions instead? If that is the way then remove the code which should no longer be used

File: tests/Test.Microsoft.Identity.Core.UIAutomation/UserInformationFieldIds.cs [hunk 5]
```diff
@@ -38,9 +38,22 @@ public void DetermineFieldIds(LabUser user)
         {
             if (user.IsFederated)
             {
-                // We use the same IDs for ADFSv3 and ADFSv4
-                PasswordInputId = CoreUiTestConstants.AdfsV4WebPasswordID;
-                SignInButtonId = CoreUiTestConstants.AdfsV4WebSubmitID;
+                switch (user.FederationProvider)
+                {
+                    case FederationProvider.AdfsV3:
+                    case FederationProvider.AdfsV4:
```
Comment (henrik-me) [inline tests/Test.Microsoft.Identity.Core.UIAutomation/UserInformationFieldIds.cs:44]: how about adding ADFS 2019 both here and in MSAL?

**Review by henrik-me — APPROVED** [review]
:shipit:

File: tests/Test.Microsoft.Identity.Core.UIAutomation/CoreMobileTestHelper.cs (hunk 4 above)
Comment (trwalke) [reply tests/Test.Microsoft.Identity.Core.UIAutomation/CoreMobileTestHelper.cs]: This is no longer needed. Reverted

File: tests/Test.Microsoft.Identity.Core.UIAutomation/UserInformationFieldIds.cs (hunk 5 above)
Comment (trwalke) [reply tests/Test.Microsoft.Identity.Core.UIAutomation/UserInformationFieldIds.cs:44]: done

File: tests/Test.ADAL.Integration/Infrastructure/SingleMessageTcpListener.cs (hunk 3 above)
Comment (trwalke) [reply tests/Test.ADAL.Integration/Infrastructure/SingleMessageTcpListener.cs]: dead code. removed

File: devApps/XFormsApp/SecondPage.cs (hunk 1 above)
Comment (trwalke) [reply devApps/XFormsApp/SecondPage.cs]: it is, this change was made for testing. it has been reverted.

**Pull request merged**